playground/minify/__tests__/minify.spec.ts

```diff
@@ -1,21 +1,78 @@
   const files = fs.readdirSync(assetsDir)
 
   const jsFile = files.find((f) => f.endsWith('.js'))
-  const jsContent = readFile(path.resolve(assetsDir, jsFile))
+  const jsContent = readFile(path.resolve(assetsDir, jsFile)).trim()
 
   const cssFile = files.find((f) => f.endsWith('.css'))
-  const cssContent = readFile(path.resolve(assetsDir, cssFile))
+  const cssContent = readFile(path.resolve(assetsDir, cssFile)).trim()
 
-  expect(jsContent).toContain('{console.log("hello world")}')
-  expect(jsContent).not.toContain('/*! explicit comment */')
+  return { jsContent, cssContent }
+}
 
-  expect(cssContent).toContain('color:#ff0000')
-  expect(cssContent).not.toContain('/*! explicit comment */')
+function getLegalFileContent(assetsDir: string) {
+  return readFile(path.resolve(assetsDir, 'LEGAL.txt'))
+}
+
+describe.runIf(isBuild)('minify', () => {
+  test('Do not preserve any legal comments', () => {
+    const { jsContent, cssContent } = getJsAndCssContent(
+      path.resolve(testDir, 'dist/none/assets'),
+    )
+
+    expect(jsContent).toContain('{console.log("hello world")}')
+    expect(jsContent).not.toContain('/*! explicit comment */')
+
+    expect(cssContent).toContain('color:#ff0000')
+    expect(cssContent).not.toContain('/*! explicit comment */')
+  })
+
+  test('Preserve legal comments', () => {
+    const { jsContent, cssContent } = getJsAndCssContent(
+      path.resolve(testDir, 'dist/inline/assets'),
+    )
+
+    expect(jsContent).toContain('/*! explicit comment */')
+    expect(cssContent).toContain('/*! explicit comment */')
+  })
+
+  test('Move all legal comments to the end of the file.', () => {
+    const { jsContent, cssContent } = getJsAndCssContent(
+      path.resolve(testDir, 'dist/eof/assets'),
+    )
+    expect(jsContent.endsWith('/*! explicit comment */')).toBeTruthy()
+    expect(cssContent.endsWith('/*! explicit comment */')).toBeTruthy()
+  })
+
+  test('Move all legal comments to a LEGAL.txt file but to not link to them.', () => {
+    const { jsContent, cssContent } = getJsAndCssContent(
+      path.resolve(testDir, 'dist/external/assets'),
+    )
+    const legaContent = getLegalFileContent(
+      path.resolve(testDir, 'dist/external'),
+    )
+    expect(jsContent).not.toContain('/*! explicit comment */')
+    expect(cssContent).not.toContain('/*! explicit comment */')
+    expect(legaContent).toContain('/*! explicit comment */')
+  })
+
+  test('Move all legal comments to a LEGAL.txt file and link to them with a comment.', () => {
+    const { jsContent, cssContent } = getJsAndCssContent(
+      path.resolve(testDir, 'dist/linked/assets'),
+    )
+    const legaContent = getLegalFileContent(
+      path.resolve(testDir, 'dist/linked'),
+    )
+    expect(jsContent).not.toContain('/*! explicit comment */')
+    expect(jsContent).toContain(
+      '/*! For license information please see LEGAL.txt */',
+    )
+    expect(cssContent).not.toContain('/*! explicit comment */')
```

Comment:
I think we want to add `/*! For license information please see LEGAL.txt */` to the CSS as well.

Reply:
Sounds reasonable.